assets/js/components/event/PresenterArea.tsx

```diff
@@ -0,0 +1,129 @@
+      webrtcConnecting = true;
+      connectWebrtc(eventChannel, client, presenters, setPresenters).then((value) => {
+        webrtc = value;
+        setIsControlPanelAvailable(true);
+        webrtcConnecting = false;
+      });
+    } else if (webrtc != null && !isClientPresenter) {
+      leaveWebrtc(webrtc, client, eventChannel);
+      webrtc = null;
+      setIsControlPanelAvailable(false);
+    }
+  }, [presenters, isClientPresenting]);
+
+  useEffect(() => {
+    askForPermissions();
+    presenterArea[client.email] = new MediaStream();
+  }, []);
+
+  useEffect(() => {
+    syncPresenters(eventChannel, setPresenters);
+  }, [eventChannel]);
+
+  const onPresenterReady = () => {
+    setIsClientPresenting(true);
+
+    eventChannel.push("presenter_ready", {
+      email: client.email,
+    });
+  };
+
+  useEffect(() => {
+    Object.values(presenters).forEach((presenter) => {
+      if (presenter.status == "connecting" && presenter.connectCallbacks.length > 0) {
+        const playerCallback = playerCallbacks[presenter.email];
+        presenter.connectCallbacks.forEach((callback) => callback(playerCallback));
+        setPresenters({
+          ...presenters,
+          [presenter.email]: { ...presenter, connectCallbacks: [] },
+        });
+      }
+    });
+  }, [presenters]);
+
+  const visiblePresenters = Object.values(presenters).filter(
+    (presenter) => presenter.status != "idle" || presenter.email == client.email
+  );
+
+  return client.email in presenters ? (
+    <div className={`PresenterArea ${mode == "hls" ? "Hidden" : ""}`}>
+      {isClientPresenting ? (
+        <div className={`StreamsGrid Grid${visiblePresenters.length}`}>
+          {visiblePresenters.map((presenter) => {
+            return (
+              <RtcPlayer
+                isMyself={client.email == presenter.email}
+                presenter={presenter}
+                playerCallbacks={playerCallbacks}
+                key={presenter.email}
+              />
+            );
+          })}
+        </div>
+      ) : (
+        <RtcPlayer
+          isMyself={true}
+          presenter={
+            Object.values(presenters).find((presenter) => presenter.email == client.email) || {
+              name: client.name,
+              email: client.email,
+              status: "idle",
+              connectCallbacks: [],
+            }
+          }
+          playerCallbacks={playerCallbacks}
+          key={client.email}
+        />
+      )}
+      {isControlPanelAvailable && (
+        <ControlPanel
+          client={client}
+          webrtc={webrtc}
+          eventChannel={eventChannel}
+          playerCallback={playerCallbacks[client.email]}
+          setMode={setMode}
+        />
+      )}
+      {!isClientPresenting && (
+        <button className="StartPresentingButton" onClick={onPresenterReady}>
+          Start presenting
+        </button>
+      )}
+    </div>
```

Comment:
I fill like the presenter preview should additionally  have a title  indicating that this is a preview (like `Presenter Preview` or something like that).
Also: consider moving this part to a different function inside `PresenterArea.tsx` or about moving this part to a separate component

Reply:
I moved the RtcPlayer into a function, the preview indication is up for a debate.